### PR TITLE
[DNM] TG Slaughter Demon port

### PR DESCRIPTION
**Pull request by Aurorablade**
Ports TG Slaughter Demon by @KorPhaeron https://github.com/tgstation/-tg-station/pull/9973 (is it bad i have been stalking this for days to port it?) All credits to him for the concept and more credits i shoved in the file itself.

>New side event/antagonist. Simple animal that can use pools of blood as
>portals to move around the station. Spooky scary.

>-CtrlClick on an adjacent blood pool (or turf with blood on it) to
>"enter" (start an infinite duration jaunt
>-CtrlClick on a blood pool while jaunted to exit at that pool
>-If you're pulling a mob while you enter a pool, the mob gets pulled in
>with you, and you devour them, healing yourself
>-Night vision

>Adds an event (admin only) that spawns one.

>Adds wizard item for summoning it. Has objective to kill the wizard (and everyone else).

What I have done:
-Ported
It works(admin and wizard spawn only)
-Added a bit more blood
-Changed Verbage on some objectives
-Did a reverse animation based on the jaunt animation for the daemon so it rises out of the blood.
-Did the whole 'would you like to play this?' question for getting the antag.
-Added in a Whisper abilty so you can creepy whisper people...

What i would like to do:
-Find places to add more blood(i need suggestions)

-Tweak on suggestion to make it fit the server and not as murderboneish(though it IS A hell demon)

-Maybe tweak the wizard bit, as from what i know on paradise once the wizard is kill round is kill. (Creatively the wizard could trick a crew member into summoning it or thier apprentiace. "Here break this for me, student..")

-MORE BLOOD
-Cooldown on entering/exiting blood?
-I would love when the demon is done eating to eject the players brain and a random organ (in hopes of borg/cloneing) but don't know how to do it.
-go from ctrl click to alt click?
-Fix any glareing errors in this 
-MORE BLOOD ALL THE BLOOD

Silly bonus:
Adds in the Honk Render and Crab render (I saw the honk render on tg...and for some reason wanted one for crabs..)
I am sure these items will be used in a responisble manner by the admins...*Adminbus honks in the distance.*

And now a visual representation of this PR:
space station 13 2015-06-16 154208

### CODE REVIEW

**Comment by Robertha89** [comment]
:+1: x :100:

**Comment by KorPhaeron** [comment]
Hurray I'm famous

**Comment by Aurorablade** [comment]
yeah alot of stuff is not working like i want but hopefully someone can push me in the right direction. or use the clow method and knock me over and repeatedly pull me over a banana peel.

**Comment by Robertha89** [comment]
for blood, check the vampiric here https://github.com/ParadiseSS13/Paradise/blob/6f69d256235c95048cc735385fe437908f66baf3/code/modules/research/xenoarchaeology/finds/finds_special.dm

**Comment by Spacemanspark** [comment]
Ahahaha I was just looking at that PR.
:+1:

**Comment by Fox-McCloud** [comment]
I'm not opposed, at all to this mini-antag being ported--as a matter of fact, I encourage it. That said, as this is your second PR and you seem to need a lot of help with this, I do encourage you to perhaps tackle a smaller project first. If you really think you're up for this, that's fine, but if you think it may be too much for you, then you may want to work on a smaller port or own self-created project.

Again, not downing you at all and I'm *NOT* opposed to this making it into the codebase--it's just that this is a pretty big project fora second PR and it may be a bit overwhelming for you.

**Comment by Aurorablade** [comment]
@Fox-McCloud I understand what you are saying completely. I am pretty used to coding and i have this bad habit of jumping in head first with a new language/codebase and pushing all the shiny buttons until i figure out what they do. also just trying to keep this to acceptable standards of omg don't muderbone me bro while still be creepy. That said i was looking at your Port of revernents its giving me a bit of idea/example to go off of as this code seems to use that as a basis so thoughts so far are:
This was intended as an admin spawn only event, should i just leave it to the event panel as a weight of 0 instead of trying to figure out the one click antag?
On the spawned version rather then the vial version, should i revert BACK to not asking a ghost if they want to be the Daemon and just going 'DING YOUR A DAEMON NOW GHOSTY POTTER'?
CTRL+click or ALT+click?

I think i am close to understanding the droplimb/organ dropping stuff i just need to stare at some code bits a bit more...

And does anyone have any suggestions on how to add MORE BLOOD? the vamp statuette gives me an idea but..

**Comment by Robertha89** [comment]
gib spawner on entering and leaving blood pools, it would create lots of localised blood

**Comment by Aurorablade** [comment]
Updated commit and added a visual representation fo the PR....

File: code/modules/events/event_container.dm [hunk 1]
```diff
@@ -169,6 +169,7 @@ var/global/list/severity_to_string = list(EVENT_LEVEL_MUNDANE = "Mundane", EVENT
 		new /datum/event_meta(EVENT_LEVEL_MODERATE, "Bluespace Anomaly",		/datum/event/anomaly/anomaly_bluespace,	50,		list(ASSIGNMENT_ENGINEER = 25)),
 		new /datum/event_meta(EVENT_LEVEL_MODERATE, "Flux Anomaly",				/datum/event/anomaly/anomaly_flux,		50,		list(ASSIGNMENT_ENGINEER = 50)),
 		new /datum/event_meta(EVENT_LEVEL_MODERATE, "Gravitational Anomaly",	/datum/event/anomaly/anomaly_grav,		200),
+		new /datum/event_meta(EVENT_LEVEL_MODERATE, "Slaughter Demon",			/datum/event/spawn_slaughter,		0),
```
Comment (Fox-McCloud) [inline code/modules/events/event_container.dm]: No point to ading this if it's an admin only event; admins with necessary permissions can trigger events manually.

File: code/modules/mob/living/simple_animal/slaughter/slaughter.dm [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+	min_n2  = 0
+	max_n2  = 0
+	minbodytemp = 0
+	faction = list("slaughter")
+	attacktext = "wildly tears into"
+	maxHealth = 250
+	health = 250
+	environment_smash = 1
+	universal_understand = 1
+	melee_damage_lower = 15//Paradise port:This was also 30, but i moved it down.
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/slaughter/slaughter.dm]: I'd leave this at 30.

File: code/modules/mob/living/simple_animal/slaughter/slaughter.dm [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+	loc = newLoc
+	src.canmove = 0
+	spawn(1)
+		src.canmove = 1
+
+/obj/effect/dummy/slaughter/ex_act(severity)
+	return 1
+
+
+///obj/effect/dummy/slaughter/singularity_act(blah)//this errored on me on compile - Aurora
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/slaughter/slaughter.dm]: your branch is outdated; you need to update it--that's why this pitched a fit.

File: code/modules/mob/living/simple_animal/slaughter/slaughter.dm [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+		src.phased = FALSE
+		//Paradise port:The hallucination sounds removed from the demon as of a recent commit..i liked it though..
+		if (prob(25))
+			var/list/voice = list('sound/hallucinations/behind_you1.ogg','sound/hallucinations/im_here1.ogg','sound/hallucinations/turn_around1.ogg','sound/hallucinations/i_see_you1.ogg')
+			playsound(get_turf(src), pick(voice),50, 1, -1)
+		src.visible_message("<span class='warning'><B>The [src] rises out of [B]!</B>")
+		new /obj/effect/gibspawner/human(get_turf(src))
+		playsound(get_turf(src), 'sound/misc/exit_blood.ogg', 100, 1, -1)
+		qdel(src.holder)
+		del(animation)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/slaughter/slaughter.dm]: qdel(animation)

File: code/modules/mob/living/simple_animal/slaughter/slaughter.dm [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+				src.devoured++
+				src << "<B> [src.devoured] [src.kidnapped] [src.eating]<B>"//Debug for tracking how many is devoured, use in objective?
+				src.kidnapped = null
+				src.eating = FALSE
+				src.pulling = FALSE
+			src.notransform = 0
+			if(!(src.eating))
+				new /obj/effect/gibspawner/human(get_turf(src))///Somewhere a janitor weeps..
+				sleep(6)///wait till the anim finishs before we delet it..
+				del(animation)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/slaughter/slaughter.dm]: qdel(animation)

File: code/modules/mob/living/simple_animal/slaughter/slaughter.dm [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+			src.loc = holder
+			src.phased = TRUE
+			src.holder = holder
+
+			if(src.kidnapped)
+				src << "<B>You being to feast on [kidnapped]. You can not move while you are doing this.</B>"
+				src.visible_message("<span class='warning'>Loud eating sounds come from the blood...</B>")
+				src.eating = TRUE
+				sleep(6)///wait till the anim finishs before we delet it..
+				del(animation)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/slaughter/slaughter.dm]: qdel(animation)

File: code/modules/mob/living/simple_animal/slaughter/slaughter.dm [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+				sleep(30)
+				new /obj/effect/gibspawner/generic(get_turf(src))
+				playsound(get_turf(src),'sound/misc/Demon_consume.ogg', 100, 1)
+				sleep(30)
+				new /obj/effect/gibspawner/generic(get_turf(src))
+				playsound(get_turf(src),'sound/misc/Demon_consume.ogg', 100, 1)
+				sleep(30)
+				var/mob/living/carbon/human/nomNom = src.kidnapped
+				var/obj/item/organ/brain/brainObj = nomNom.internal_organs_by_name["brain"]
+				if(brainObj && istype(brainObj))
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/slaughter/slaughter.dm]: Bitty iff on this not being a permadeath.

File: code/modules/mob/living/simple_animal/slaughter/slaughter.dm [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+			animation.density = 0
+			animation.anchored = 1
+			animation.icon = 'icons/mob/mob.dmi'
+			animation.icon_state = "jaunt"
+			animation.layer = 5
+			animation.master = holder
+			animation.dir = src.dir
+			src.ExtinguishMob()
+			if(src.buckled)
+				src.buckled = null
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/slaughter/slaughter.dm]: src.buckled.unbuckle() instead

File: code/game/gamemodes/wizard/spellbook.dm [hunk 9]
```diff
@@ -386,6 +392,11 @@
 			src.uses++
 			qdel(O)
 
+	if(istype(O, /obj/item/weapon/antag_spawner/slaughter_demon))
+		user << "<span class='notice'>On second thought, maybe summoning a demon is a bad idea. You refund your points.</span>"
+		src.uses++
+		del(O)
```
Comment (Fox-McCloud) [inline code/game/gamemodes/wizard/spellbook.dm]: qdel(O)

File: code/modules/mob/living/simple_animal/slaughter/slaughter.dm [hunk 10]
```diff
@@ -0,0 +1,274 @@
...
+	var/playstyle_string = "<B>You are the Slaughter Demon, a terible creature from another existence. You have a single desire: To kill.  \
+						You may Ctrl+Click on blood pools to travel through them, appearing and dissaapearing from the station at will. \
+						Pulling a dead or critical mob while you enter a pool will pull them in with you, allowing you to feast. </B>"
+   //Paradise todo:change to altclick,add objectives. need ideas as i want people to be creative, not just RAWR KILL
+
+
+/mob/living/simple_animal/slaughter/New()
+	..()
+	spawn(5)
+	AddSpell(new /obj/effect/proc_holder/spell/wizard/targeted/slaughter_whisper)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/slaughter/slaughter.dm]: ehhh...not sure if this is the best way to handle it or if it should be added to the mind.

File: code/modules/mob/living/simple_animal/slaughter/slaughter.dm [hunk 11]
```diff
@@ -0,0 +1,274 @@
...
+			animation.master = holder
+			animation.dir = src.dir
+			src.ExtinguishMob()
+			if(src.buckled)
+				src.buckled = null
+			if(src.pulling)
+				if(istype(src.pulling, /mob/living/))
+					var/mob/living/victim = src.pulling
+					var/mob/living/carbon/human/victimType = src.pulling
+					if (victimType.species.flags & IS_SYNTHETIC)//IPCs are too crunchy....I am also calling this wrong.
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/slaughter/slaughter.dm]: Let's not give IPCs immunity to an ethereal demon.

File: code/modules/mob/living/simple_animal/slaughter/slaughter.dm (hunk 7 above)
Comment (Aurorablade) [inline code/modules/mob/living/simple_animal/slaughter/slaughter.dm]: @Fox-McCloud Well i wasn't sure if perma death would be alright or not. I can change it to try to throw out a random organ.

File: code/modules/mob/living/simple_animal/slaughter/slaughter.dm (hunk 10 above)
Comment (Aurorablade) [inline code/modules/mob/living/simple_animal/slaughter/slaughter.dm]: I will try adding it in to mind again.

**Comment by Aurorablade** [comment]
Progress:Added in the Spell Exsanguinate
Spell Quirk: Exanguinate and whisper only work properly when phased in. I am assuming this is due to the fact when you phase out you transfer to a holder dummy. Will need to think on this when my brain is more active.(Read...Need Caffine...)

Removed IPC immunity and put in alot of robot gibs...NOMNOMNOM CRUNCHY CIRCUITS

**Comment by Aurorablade** [comment]
DISREGARD SPELLS! USE VERBS! *CACKLES MADLY*

**Comment by Fox-McCloud** [comment]
I...what.

**Comment by Iamgoofball** [comment]
# RIP AND TEAR

## RIP AND TEAR YOUR GUTS

**Comment by Aurorablade** [comment]
@Iamgoofball well the orignal intent from TG was admin spawn only (or suicidal wizard/Wizards giving item to apprentice/Admins spawning blood vial on unsuspecting Greytide) according to @Fox-McCloud  "No point to ading this if it's an admin only event; admins with necessary permissions can trigger events manually."

I am open to suggestions for anything though.

Disclaimer: Pun Pun WAS harmed mutliple times in my next commit (And i LIKE pun pun)

**Comment by KorPhaeron** [comment]
@Iamgoofball 
@Aurorablade 

I wanted to test it first to see if it was suitable for being a normal random antag like ninja or aliens, so I set the event chance to zero for now

**Comment by Aurorablade** [comment]
@KorPhaeron I hope you don't mind me running away with this adding things...

**Comment by Aurorablade** [comment]
think i should borrow the fluff objective code from https://github.com/ParadiseSS13/Paradise/pull/1326 ? if so what would people like to see?

...And should this thing interact with the chaplain at all?

**Comment by KorPhaeron** [comment]
Chaplain already interacts with cult, this thing interacts with janitors.

And no I don't mind you adding stuff, its the nature of open source.

Eating X number of people would be a good objective, or eating certain people.

**Comment by Aurorablade** [comment]
latest commit:
Changed the GibEat() proc to use the mobs blood color. (defualts to human gibs if we eat Ian)
Added in (or tried to) so gib and innard throwing.
TRIED TO make sure we cannot sink into robot debris.

Reworked OBjectives slightly.

Should I consider not letting this thing EAT DEAD CYBORGS?!

**Comment by Aurorablade** [comment]
so may add fluff objectives such as kill only this role, kill the janitor..or otherwise..Anyone else got fun suggestions for anything?

**Comment by Aurorablade** [comment]
Ability Brainstorm, by suggestion: Devour so many people, Suddenly able to throw gibs at people while phased in/out...Just to make them bloody...and the shock of have gibs THROWN AT YOU

**Comment by Aurorablade** [comment]
Added in a throwable gib move when you eat give people. 1 minute CD if i did it right. MAY need to tweak the throw_at speed...cuase at one point..I killed pun pun..with gibs....but then again i also had the CD off for testing...so i was spam throwing gibs...

**Comment by Aurorablade** [comment]
Oh for the love of...BRB Making new branch

**Pull request closed without merging**